app/models/deletion.rb

```diff
@@ -7,12 +7,18 @@ class Deletion < ActiveRecord::Base
 
   before_validation :record_metadata
   after_create :remove_from_index, :set_yanked_info_checksum
-  after_commit :remove_from_storage
+  after_commit :remove_from_storage, on: :create
   after_commit :expire_cache
   after_commit :update_search_index
 
   attr_accessor :version
 
+  def restore!
+    restore_to_index
+    restore_to_storage
+    destroy!
```

Comment:
> Can you please explain what is the value of marking it restored instead of deleting the record?

The value, would be to have a breadcrumb if deleted/restored gems, for future data finds.
However, I don't know how much do we care about that or not.


I agree that adding that flag would add more complexity to the code.